src/firebase/auth/index.ts

```diff
@@ -23,15 +23,13 @@ export const auth = getAuth(app)
  * @param {string} email - new user's e-mail;
  * @param {string} password - new user's password;
  */
-export const createNewUser = async ({ firstName, lastName, role, email, password }: INewUserProps): Promise<void> => {
+export const createNewUser = async ({ firstName, lastName, email, password }: INewUserProps): Promise<void> => {
 	try {
-		const newUser = await createUserWithEmailAndPassword(auth, email, password)
+		const newUser = await createUserWithEmailAndPassword(auth, email as string, password as string)
```

Comment:
I suspect `INewUserProps` has `email` and `password` as optional props, because why else would you need to specify that it's a string here.

So either the `INewUserProps` should have email and password as a required props, or you simply can add an `if` statement that checks those two.

Reply:
Hope, that i undersatsdnd you in right way